src/rules/arrayTypeRule.ts

```diff
@@ -51,83 +53,116 @@ export class Rule extends Lint.Rules.AbstractRule {
+        const parens = typeNeedsParentheses(type);
+        const fix = ctx.createFix(
+            // Delete 'Array<'
+            Lint.Replacement.replaceFromTo(node.getStart(), type.getStart(), parens ? "(" : ""),
+            // Delete '>' and replace with '[]
+            Lint.Replacement.replaceFromTo(type.getEnd(), node.getEnd(), parens ? ")[]" : "[]"));
+        ctx.addFailureAtNode(node, failureString, fix);
     }
+}
 
-    private isSimpleType(nodeType: ts.TypeNode) {
-        switch (nodeType.kind) {
-            case ts.SyntaxKind.AnyKeyword:
-            case ts.SyntaxKind.ArrayType:
-            case ts.SyntaxKind.BooleanKeyword:
-            case ts.SyntaxKind.NullKeyword:
-            case ts.SyntaxKind.UndefinedKeyword:
-            case ts.SyntaxKind.NumberKeyword:
-            case ts.SyntaxKind.StringKeyword:
-            case ts.SyntaxKind.SymbolKeyword:
-            case ts.SyntaxKind.VoidKeyword:
-            case ts.SyntaxKind.NeverKeyword:
+function typeNeedsParentheses(type: ts.TypeNode): boolean {
+    switch (type.kind) {
+        case ts.SyntaxKind.UnionType:
+        case ts.SyntaxKind.FunctionType:
+        case ts.SyntaxKind.IntersectionType:
+        case ts.SyntaxKind.TypeOperator:
+            return true;
+        default:
+            return false;
+    }
+}
+
+function isArrayIdentifier(name: ts.EntityName) {
+    return name.kind === ts.SyntaxKind.Identifier && name.text === "Array";
+}
+
+function isSimpleType(nodeType: ts.TypeNode): boolean {
+    switch (nodeType.kind) {
+        case ts.SyntaxKind.AnyKeyword:
+        case ts.SyntaxKind.ArrayType:
+        case ts.SyntaxKind.BooleanKeyword:
+        case ts.SyntaxKind.NullKeyword:
+        case ts.SyntaxKind.ObjectKeyword:
+        case ts.SyntaxKind.UndefinedKeyword:
+        case ts.SyntaxKind.NumberKeyword:
+        case ts.SyntaxKind.StringKeyword:
+        case ts.SyntaxKind.SymbolKeyword:
+        case ts.SyntaxKind.VoidKeyword:
+        case ts.SyntaxKind.NeverKeyword:
+            return true;
+        case ts.SyntaxKind.TypeReference:
+            // TypeReferences must be non-generic or be another Array with a simple type
+            const { typeName, typeArguments } = nodeType as ts.TypeReferenceNode;
+            if (!typeArguments) {
                 return true;
-            case ts.SyntaxKind.TypeReference:
-                // TypeReferences must be non-generic or be another Array with a simple type
-                const node = nodeType as ts.TypeReferenceNode;
-                const typeArgs = node.typeArguments;
-                if (!typeArgs || typeArgs.length === 0 || node.typeName.getText() === "Array" && this.isSimpleType(typeArgs[0])) {
+            }
+            switch (typeArguments.length) {
+                case 0:
                     return true;
-                } else {
+                case 1:
+                    return utils.isIdentifier(typeName) && typeName.text === "Array" && isSimpleType(typeArguments[0]);
```

Comment:
use `typeName.kind === ts.SyntaxKind.Identifier` instead of `isIdentifier`. `typeName` is a discriminated union so this narrows the type.